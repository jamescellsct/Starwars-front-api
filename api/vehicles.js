import express from "express";

const router = express.Router();

var vehicles = require("./fakeData/db.json").vehicles;

router.get("/vehicles", function(req, res) {
  return res.status(200).json({
    count: vehicles.count,
    results: vehicles.results[0]
  });
});

router.get("/vehicles/:page", function(req, res) {
  let { page } = req.params;
  page = page.slice(5, 6) - 1;

  if (!vehicles.results[page]) {
    return res.status(404).send({
      error: {
        message: "Page Not Found!"
      }
    });
  }

  return res.status(200).json({
    count: vehicles.count,
    results: vehicles.results[page]
  });
});

export default router;
