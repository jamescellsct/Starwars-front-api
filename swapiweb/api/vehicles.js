import express from "express";
const router = express.Router();

var vehicles = require("./fakeData/db.json").vehicles;

router.get("/vehicles", function(req, res) {
  return res.status(200).json({
    data: vehicles
  });
});

export default router;
