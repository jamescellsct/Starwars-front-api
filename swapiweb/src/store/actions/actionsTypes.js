//Peoples Actions Types
const FETCH_PEOPLE_STARTED = "FETCH_PEOPLE_STARTED";
const FETCH_PEOPLE_SUCCEEDED = "FETCH_PEOPLE_SUCCEEDED";
const FETCH_PEOPLE_FAILED = "FETCH_PEOPLE_FAILED";
const UPDATE_PEOPLE_IS_FETCHING = "UPDATE_PEOPLE_IS_FETCHING";
const UPDATE_PEOPLE_SUCCEEDED = "UPDATE_PEOPLE_SUCCEEDED";
const UPDATE_PEOPLE_FAILED = "UPDATE_PEOPLE_FAILED";
export const peoplesAction = {
  FETCH_PEOPLE_STARTED,
  FETCH_PEOPLE_SUCCEEDED,
  FETCH_PEOPLE_FAILED,
  UPDATE_PEOPLE_IS_FETCHING,
  UPDATE_PEOPLE_SUCCEEDED,
  UPDATE_PEOPLE_FAILED
};

//Planets Actions Types
const FETCH_PLANETS_STARTED = "FETCH_PLANETS_STARTED";
const FETCH_PLANETS_SUCCEEDED = "FETCH_PLANETS_SUCCEEDED";
const FETCH_PLANETS_FAILED = "FETCH_PLANETS_FAILED";
const UPDATE_PLANETS_IS_FETCHING = "UPDATE_PLANETS_IS_FETCHING";
const UPDATE_PLANETS_SUCCEEDED = "UPDATE_PLANETS_SUCCEEDED";
const UPDATE_PLANETS_FAILED = "UPDATE_PLANETS_FAILED";
export const planetsAction = {
  FETCH_PLANETS_STARTED,
  FETCH_PLANETS_SUCCEEDED,
  FETCH_PLANETS_FAILED,
  UPDATE_PLANETS_IS_FETCHING,
  UPDATE_PLANETS_SUCCEEDED,
  UPDATE_PLANETS_FAILED
};

//Starships Actions Types
const FETCH_STARSHIPS_STARTED = "FETCH_STARSHIPS_STARTED";
const FETCH_STARSHIPS_SUCCEEDED = "FETCH_STARSHIPS_SUCCEEDED";
const FETCH_STARSHIPS_FAILED = "FETCH_STARSHIPS_FAILED";
const UPDATE_STARSHIPS_IS_FETCHING = "UPDATE_STARSHIPS_IS_FETCHING";
const UPDATE_STARSHIPS_SUCCEEDED = "UPDATE_STARSHIPS_SUCCEEDED";
const UPDATE_STARSHIPS_FAILED = "UPDATE_STARSHIPS_FAILED";
export const starshipsAction = {
  FETCH_STARSHIPS_STARTED,
  FETCH_STARSHIPS_SUCCEEDED,
  FETCH_STARSHIPS_FAILED,
  UPDATE_STARSHIPS_IS_FETCHING,
  UPDATE_STARSHIPS_SUCCEEDED,
  UPDATE_STARSHIPS_FAILED
};

//Vehicles Actions Types
const FETCH_VEHICLES_STARTED = "FETCH_VEHICLES_STARTED";
const FETCH_VEHICLES_SUCCEEDED = "FETCH_VEHICLES_SUCCEEDED";
const FETCH_VEHICLES_FAILED = "FETCH_VEHICLES_FAILED";
const UPDATE_VEHICLES_IS_FETCHING = "UPDATE_VEHICLES_IS_FETCHING";
const UPDATE_VEHICLES_SUCCEEDED = "UPDATE_VEHICLES_SUCCEEDED";
const UPDATE_VEHICLES_FAILED = "UPDATE_VEHICLES_FAILED";
export const vehiclesAction = {
  FETCH_VEHICLES_STARTED,
  FETCH_VEHICLES_SUCCEEDED,
  FETCH_VEHICLES_FAILED,
  UPDATE_VEHICLES_IS_FETCHING,
  UPDATE_VEHICLES_SUCCEEDED,
  UPDATE_VEHICLES_FAILED
};

//Pagination Action Types
const PAGE_SELECTED = "PAGE_SELECTED";
export const paginationAction = {
  PAGE_SELECTED
};
