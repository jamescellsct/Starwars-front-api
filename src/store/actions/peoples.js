import { peoplesAction } from "./actionsTypes";
import { getPeople } from "../../services/api/peopleService";
import { handleResulstWithId } from "../../utils/handleResultsWithId";

const fetchPeopleStarted = () => ({
  type: peoplesAction.FETCH_PEOPLE_STARTED
});

const fetchPeopleSucceeded = people => ({
  type: peoplesAction.FETCH_PEOPLE_SUCCEEDED,
  people
});

const fetchPeopleFailed = error => ({
  type: peoplesAction.FETCH_PEOPLE_FAILED,
  error
});

export const fetchPeople = () => dispatch => {
  dispatch(fetchPeopleStarted());

  return getPeople()
    .then(response => {
      const { results, count } = response;
      dispatch(
        fetchPeopleSucceeded({
          page: 1,
          results: handleResulstWithId(results, 1),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(fetchPeopleFailed(error));
    });
};

const updatePeopleIsFetching = () => ({
  type: peoplesAction.UPDATE_PEOPLE_IS_FETCHING
});

const updatePeopleSucceeded = people => ({
  type: peoplesAction.UPDATE_PEOPLE_SUCCEEDED,
  people
});

const updatePeopleFailed = error => ({
  type: peoplesAction.UPDATE_PEOPLE_FAILED,
  error
});

export const updatePeople = page => dispatch => {
  dispatch(updatePeopleIsFetching());

  getPeople(page)
    .then(response => {
      const { results, count } = response;
      dispatch(
        updatePeopleSucceeded({
          page,
          results: handleResulstWithId(results, page),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(updatePeopleFailed(error));
    });
};
