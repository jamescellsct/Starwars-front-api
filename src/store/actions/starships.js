import { starshipsAction } from "./actionsTypes";
import { getStarships } from "../../services/api/starshipsService";
import { handleResulstWithId } from "../../utils/handleResultsWithId";

const fetchStarshipsStarted = () => ({
  type: starshipsAction.FETCH_STARSHIPS_STARTED
});

const fetchStarshipsSucceeded = starships => ({
  type: starshipsAction.FETCH_STARSHIPS_SUCCEEDED,
  starships
});

const fetchStarshipsFailed = error => ({
  type: starshipsAction.FETCH_STARSHIPS_FAILED,
  error
});

export const fetchStarships = () => dispatch => {
  dispatch(fetchStarshipsStarted());

  return getStarships()
    .then(response => {
      const { results, count } = response;
      dispatch(
        fetchStarshipsSucceeded({
          page: 1,
          results: handleResulstWithId(results, 1),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(fetchStarshipsFailed(error));
    });
};

const updateStarshipsIsFetching = () => ({
  type: starshipsAction.UPDATE_STARSHIPS_IS_FETCHING
});

const updateStarshipsSucceeded = starships => ({
  type: starshipsAction.UPDATE_STARSHIPS_SUCCEEDED,
  starships
});

const updateStarshipsFailed = error => ({
  type: starshipsAction.UPDATE_STARSHIPS_FAILED,
  error
});

export const updateStarships = page => dispatch => {
  dispatch(updateStarshipsIsFetching());
  return getStarships(page)
    .then(response => {
      const { results, count } = response;
      dispatch(
        updateStarshipsSucceeded({
          page,
          results: handleResulstWithId(results, page),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(updateStarshipsFailed(error));
    });
};
