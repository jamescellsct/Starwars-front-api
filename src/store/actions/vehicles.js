import { vehiclesAction } from "./actionsTypes";
import { getVehicles } from "../../services/api/vehiclesService";
import { handleResulstWithId } from "../../utils/handleResultsWithId";

const fetchVehiclesStarted = () => ({
  type: vehiclesAction.FETCH_VEHICLES_STARTED
});

const fetchVehiclesSucceeded = vehicles => ({
  type: vehiclesAction.FETCH_VEHICLES_SUCCEEDED,
  vehicles
});

const fetchVehiclesFailed = error => ({
  type: vehiclesAction.FETCH_VEHICLES_FAILED,
  error
});

export const fetchVehicles = () => dispatch => {
  dispatch(fetchVehiclesStarted());

  return getVehicles()
    .then(response => {
      const { results, count } = response;
      dispatch(
        fetchVehiclesSucceeded({
          page: 1,
          results: handleResulstWithId(results, 1),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(fetchVehiclesFailed(error));
    });
};

const updateVehiclesIsFetching = () => ({
  type: vehiclesAction.UPDATE_VEHICLES_IS_FETCHING
});

const updateVehiclesSucceeded = vehicles => ({
  type: vehiclesAction.UPDATE_VEHICLES_SUCCEEDED,
  vehicles
});

const updateVehiclesFailed = error => ({
  type: vehiclesAction.UPDATE_VEHICLES_FAILED,
  error
});

export const updateVehicles = page => dispatch => {
  dispatch(updateVehiclesIsFetching());
  return getVehicles(page)
    .then(response => {
      const { results, count } = response;
      dispatch(
        updateVehiclesSucceeded({
          page,
          results: handleResulstWithId(results, page),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(updateVehiclesFailed(error));
    });
};
