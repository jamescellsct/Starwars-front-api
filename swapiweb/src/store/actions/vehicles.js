import { vehiclesAction } from "./actionsTypes";
import { swapiRequest } from "../../services/swapi_connect";

const fetchVehiclesStarted = () => ({
  type: vehiclesAction.FETCH_VEHICLES_STARTED
});

const fetchVehiclesSucceeded = vehicles => ({
  type: vehiclesAction.FETCH_VEHICLES_SUCCEEDED,
  vehicles
});

const fetchVehiclesFailed = error => ({
  type: vehiclesAction.FETCH_VEHICLES_FAILED,
  error
});

export const fetchVehicles = () => dispatch => {
  dispatch(fetchVehiclesStarted());

  swapiRequest(null, null, "vehicles")
    .then(response => {
      dispatch(
        fetchVehiclesSucceeded({
          results: response.results,
          page: response.page,
          pageCount: response.pageCount
        })
      );
    })
    .catch(error => {
      dispatch(fetchVehiclesFailed(error));
    });
};

const updateVehiclesIsFetching = () => ({
  type: vehiclesAction.UPDATE_VEHICLES_IS_FETCHING
});

const updateVehiclesSucceeded = vehicles => ({
  type: vehiclesAction.UPDATE_VEHICLES_SUCCEEDED,
  vehicles
});

const updateVehiclesFailed = error => ({
  type: vehiclesAction.UPDATE_VEHICLES_FAILED,
  error
});

export const updateVehicles = page => dispatch => {
  dispatch(updateVehiclesIsFetching());
  swapiRequest(null, page, "vehicles")
    .then(response => {
      dispatch(
        updateVehiclesSucceeded({
          results: response.results,
          page: response.page,
          pageCount: response.pageCount
        })
      );
    })
    .catch(error => {
      dispatch(updateVehiclesFailed(error));
    });
};
