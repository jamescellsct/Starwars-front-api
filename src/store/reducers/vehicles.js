import { vehiclesAction } from "../actions/actionsTypes";

export default function vehicles(state = null, action) {
  switch (action.type) {
    case vehiclesAction.FETCH_VEHICLES_STARTED:
      return Object.assign({}, state, {
        isFetching: true
      });

    case vehiclesAction.FETCH_VEHICLES_SUCCEEDED:
      return Object.assign({}, state, {
        pages: [
          ...(state.pages || []),
          { page: action.vehicles.page, list: action.vehicles.results }
        ],
        pageCount: action.vehicles.pageCount,
        isFetching: false
      });

    case vehiclesAction.FETCH_VEHICLES_FAILED:
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });

    case vehiclesAction.UPDATE_VEHICLES_IS_FETCHING:
      return Object.assign({}, state, {
        isFetching: true
      });
    case vehiclesAction.UPDATE_VEHICLES_SUCCEEDED:
      return Object.assign({}, state, {
        pages: [
          ...state.pages,
          { page: action.vehicles.page, list: action.vehicles.results }
        ],
        pageCount: action.vehicles.pageCount,
        isFetching: false
      });

    case vehiclesAction.UPDATE_VEHICLES_FAILED:
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });

    default:
      return state;
  }
}
