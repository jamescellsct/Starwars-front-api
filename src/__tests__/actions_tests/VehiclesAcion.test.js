import thunk from "redux-thunk";
import configureMockStore from "redux-mock-store";
import nock from "nock";

import { vehiclesAction } from "../../store/actions/actionsTypes";
import { fetchVehicles } from "../../store/actions/vehicles";
import BASE_URL from "./baseUrl";

const midlewares = [thunk];
const mockStore = configureMockStore(midlewares);

const mockResults = [
  {
    name: "Blanche Gilbert",
    avatar: "http://i.pravatar.cc/40?img=0.7765365288473662",
    model: "liaj",
    manufacturer: "demat",
    length: 917,
    max_atmosphering_speed: 532,
    passengers: 1856,
    cargo_capacity: 3015845,
    consumables: 1,
    vehicle_class: "urdo",
    idItem: 1
  },
  {
    name: "Gavin George",
    avatar: "http://i.pravatar.cc/40?img=0.10192561403480482",
    model: "suwfena",
    manufacturer: "ca",
    length: 328,
    max_atmosphering_speed: 418,
    passengers: 164,
    cargo_capacity: 3645260,
    consumables: 4,
    vehicle_class: "nopedhus",
    idItem: 2
  },
  {
    name: "Edward Cruz",
    avatar: "http://i.pravatar.cc/40?img=0.11335424551707507",
    model: "az",
    manufacturer: "zepceb",
    length: 810,
    max_atmosphering_speed: 674,
    passengers: 1728,
    cargo_capacity: 5694589,
    consumables: 4,
    vehicle_class: "lu",
    idItem: 3
  }
];

describe("Testing Vehicles Action", () => {
  it("Shold create an action to fetching vehicles data", () => {
    afterEach(() => {
      nock.cleanAll();
    });

    nock(BASE_URL)
      .defaultReplyHeaders({ "access-control-allow-origin": "*" })
      .get("/vehicles")
      .reply(200, {
        count: 20,
        results: mockResults
      });

    const expectedAction = [
      { type: vehiclesAction.FETCH_VEHICLES_STARTED },
      {
        type: vehiclesAction.FETCH_VEHICLES_SUCCEEDED,
        vehicles: {
          page: 1,
          results: mockResults,
          pageCount: 2
        }
      }
    ];

    const store = mockStore({ vehicles: {} });
    return store.dispatch(fetchVehicles()).then(() => {
      expect(store.getActions()).toEqual(expectedAction);
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
