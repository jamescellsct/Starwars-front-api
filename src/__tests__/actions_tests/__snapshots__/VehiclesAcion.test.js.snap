// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Vehicles Action Shold create an action to fetching vehicles data 1`] = `
Array [
  Object {
    "type": "FETCH_VEHICLES_STARTED",
  },
  Object {
    "type": "FETCH_VEHICLES_SUCCEEDED",
    "vehicles": Object {
      "page": 1,
      "pageCount": 2,
      "results": Array [
        Object {
          "avatar": "http://i.pravatar.cc/40?img=0.7765365288473662",
          "cargo_capacity": 3015845,
          "consumables": 1,
          "idItem": 1,
          "length": 917,
          "manufacturer": "demat",
          "max_atmosphering_speed": 532,
          "model": "liaj",
          "name": "Blanche Gilbert",
          "passengers": 1856,
          "vehicle_class": "urdo",
        },
        Object {
          "avatar": "http://i.pravatar.cc/40?img=0.10192561403480482",
          "cargo_capacity": 3645260,
          "consumables": 4,
          "idItem": 2,
          "length": 328,
          "manufacturer": "ca",
          "max_atmosphering_speed": 418,
          "model": "suwfena",
          "name": "Gavin George",
          "passengers": 164,
          "vehicle_class": "nopedhus",
        },
        Object {
          "avatar": "http://i.pravatar.cc/40?img=0.11335424551707507",
          "cargo_capacity": 5694589,
          "consumables": 4,
          "idItem": 3,
          "length": 810,
          "manufacturer": "zepceb",
          "max_atmosphering_speed": 674,
          "model": "az",
          "name": "Edward Cruz",
          "passengers": 1728,
          "vehicle_class": "lu",
        },
      ],
    },
  },
]
`;

exports[`Testing Vehicles Action Should create an action to update vehicles data 1`] = `
Array [
  Object {
    "type": "UPDATE_VEHICLES_IS_FETCHING",
  },
  Object {
    "type": "UPDATE_VEHICLES_SUCCEEDED",
    "vehicles": Object {
      "page": 1,
      "pageCount": 2,
      "results": Array [
        Object {
          "avatar": "http://i.pravatar.cc/40?img=0.7765365288473662",
          "cargo_capacity": 3015845,
          "consumables": 1,
          "idItem": 1,
          "length": 917,
          "manufacturer": "demat",
          "max_atmosphering_speed": 532,
          "model": "liaj",
          "name": "Updating Gilbert",
          "passengers": 1856,
          "vehicle_class": "urdo",
        },
        Object {
          "avatar": "http://i.pravatar.cc/40?img=0.10192561403480482",
          "cargo_capacity": 3645260,
          "consumables": 4,
          "idItem": 2,
          "length": 328,
          "manufacturer": "ca",
          "max_atmosphering_speed": 418,
          "model": "suwfena",
          "name": "Updating George",
          "passengers": 164,
          "vehicle_class": "nopedhus",
        },
        Object {
          "avatar": "http://i.pravatar.cc/40?img=0.11335424551707507",
          "cargo_capacity": 5694589,
          "consumables": 4,
          "idItem": 3,
          "length": 810,
          "manufacturer": "zepceb",
          "max_atmosphering_speed": 674,
          "model": "az",
          "name": "Edward Cruz",
          "passengers": 1728,
          "vehicle_class": "lu",
        },
      ],
    },
  },
]
`;

exports[`Testing Vehicles Action Should return ERROR in fetching vehicles data 1`] = `
Array [
  Object {
    "type": "FETCH_VEHICLES_STARTED",
  },
  Object {
    "error": [Error: Request failed with status code 400],
    "type": "FETCH_VEHICLES_FAILED",
  },
]
`;

exports[`Testing Vehicles Action Should return ERROR on updating vehicles data 1`] = `
Array [
  Object {
    "type": "UPDATE_VEHICLES_IS_FETCHING",
  },
  Object {
    "error": [Error: Request failed with status code 400],
    "type": "UPDATE_VEHICLES_FAILED",
  },
]
`;
